cmd/clusterctl/cmd/rollout/pause.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rollout
+
+import (
+	"github.com/spf13/cobra"
+	"k8s.io/kubectl/pkg/util/templates"
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/client"
+)
+
+// pauseOptions is the start of the data required to perform the operation.
+type pauseOptions struct {
+	kubeconfig        string
+	kubeconfigContext string
+	resources         []string
+	namespace         string
+}
+
+var pauseOpt = &pauseOptions{}
+
+var (
+	pauseLong = templates.LongDesc(`
+		pause of cluser-api resources.
+
+	        Any new updates to the resource will not have an effect as long as the deployment is paused`)
```

Comment:
Pulled this from the `kubectl rollout pause -h`
```suggestion
		Mark the provided cluster-api resource as paused.

	        Paused resources will not be reconciled by a controller. Use "clusterctl alpha rollout resume" to resume a paused resource. Currently only MachineDeployments support being paused.`)
```